src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Navigation;
+using OrchardCore.Settings;
+using OrchardCore.Taxonomies.Indexing;
+using OrchardCore.Taxonomies.Models;
+using OrchardCore.Taxonomies.ViewModels;
+using YesSql;
+
+namespace OrchardCore.Taxonomies.Drivers
+{
+    public class TermPartContentDriver : ContentDisplayDriver
+    {
+        private readonly ISession _session;
+        private readonly ISiteService _siteService;
+        private readonly IContentManager _contentManager;
+
+        public TermPartContentDriver(
+            ISession session,
+            ISiteService siteService,
+            IContentManager contentManager)
+        {
+            _session = session;
+            _siteService = siteService;
+            _contentManager = contentManager;
+        }
+
+        public override Task<IDisplayResult> DisplayAsync(ContentItem model, BuildDisplayContext context)
+        {
+            var part = model.As<TermPart>();
+            if (part != null)
+            {
+                return Task.FromResult<IDisplayResult>(Initialize<TermPartViewModel>("TermPart", async m =>
+                {
+                    var siteSettings = await _siteService.GetSiteSettingsAsync();
+                    var pager = await GetPagerAsync(context.Updater, siteSettings.PageSize);
+                    var jTerms = model.Content.Terms as JArray;
+
+                    m.TermPart = part;
+                    m.ContentItems = (await QueryTermItemsAsync(part, pager)).ToArray();
+                    m.Context = context;
+                    m.Terms = TermDriverHelper.PopulateTermEntries(jTerms?.ToObject<List<ContentItem>>(), context);
+                    m.Pager = await context.New.PagerSlim(pager);
+                })
+                .Location("Detail", "Content:5"));
+            }
+
```

Comment:
`ContentDriver` not `Part` driver as the `TermPart` is welded.

Paged in the same way the `ListPart` is paged.

Includes both related content items, and child terms (not the entire set, just children at the current level of the term heirachy